File: tests/DeckTest.php
<?php

class DeckTest extends PHPUnit_Framework_TestCase
{
    /**
     * @var Deck
     */
    private $_Deck;

    protected function setUp()
    {
        $this->_Deck = new Deck();
    }

    /**
     * @test
     */
    public function anEmpyDeckShouldHaveASizeOfZero()
    {
        $this->assertEquals(0, $this->_Deck->size());
    }

    /**
     * @test
     */
    public function addingACardShouldIncreaseTheSizeByOne()
    {
        $this->_Deck->add(Card::aceOf(Suit::Spades()));
        $this->assertEquals(1, $this->_Deck->size());
    }

    /**
     * @test
     */
    public function addingACardTwiceShouldCauseADeckExceptionToBeThrown()
    {
        $this->_Deck->add(Card::twoOf(Suit::Hearts()));

        $this->setExpectedException('DeckIntegrityException', 'Cannot contain the same card twice!');
        $this->_Deck->add(Card::twoOf(Suit::Hearts()));
    }

    /**
     * @test
     */
    public function populateShouldFillTheDeckWithAll52Cards()
    {
        $ExpectedCards = array();

        $Suits = array(
            Suit::Spades(),
            Suit::Hearts(),
            Suit::Clubs(),
            Suit::Diamonds(),
        );

        foreach ($Suits as $Suit) {
            $ExpectedCards[] = Card::aceOf($Suit);
            $ExpectedCards[] = Card::twoOf($Suit);
            $ExpectedCards[] = Card::threeOf($Suit);
            $ExpectedCards[] = Card::fourOf($Suit);
            $ExpectedCards[] = Card::fiveOf($Suit);
            $ExpectedCards[] = Card::sixOf($Suit);
            $ExpectedCards[] = Card::sevenOf($Suit);
            $ExpectedCards[] = Card::eightOf($Suit);
            $ExpectedCards[] = Card::nineOf($Suit);
            $ExpectedCards[] = Card::tenOf($Suit);
            $ExpectedCards[] = Card::jackOf($Suit);
            $ExpectedCards[] = Card::queenOf($Suit);
            $ExpectedCards[] = Card::kingOf($Suit);
        }

        $this->_Deck->populate();

        $this->assertEquals($ExpectedCards, $this->_Deck->getCards());
    }
}

File: tests/DeckTest.cs
using System.Linq;
using NUnit.Framework;
using System.Collections.Generic;
using PokerSharp;

[TestFixture]
class DeckTest {

    private Deck Deck;

    [SetUp]
    protected void setUp() {
        Deck = new Deck();
    }

    [Test]
    public void anEmptyDeckShouldHaveASizeOfZero() {
        Assert.AreEqual(0, Deck.size());
    }

    [Test]
    public void addingACardShouldIncreaseTheSizeByOne() {
        Deck.add(Cards.aceOf(Suit.Spades()));
        Assert.AreEqual(1, Deck.size());
    }

    [Test]
    [ExpectedException(typeof(DeckIntegrityException))]
    public void addingACardTwiceShouldCauseADeckExceptionToBeThrown() {
        Deck.add(Cards.twoOf(Suit.Hearts()));
        Deck.add(Cards.twoOf(Suit.Hearts()));
    }

    [Test]
    public void populateShouldFillTheDeckWithAll52Cards() {
        var ExpectedCards = new List<Card>();

        Suit[] Suits = {
            Suit.Spades(),
            Suit.Hearts(),
            Suit.Clubs(),
            Suit.Diamonds(),
        };

        foreach (var suit in Suits) {
            ExpectedCards.Add(Cards.aceOf(suit));
            ExpectedCards.Add(Cards.twoOf(suit));
            ExpectedCards.Add(Cards.threeOf(suit));
            ExpectedCards.Add(Cards.fourOf(suit));
            ExpectedCards.Add(Cards.fiveOf(suit));
            ExpectedCards.Add(Cards.sixOf(suit));
            ExpectedCards.Add(Cards.sevenOf(suit));
            ExpectedCards.Add(Cards.eightOf(suit));
            ExpectedCards.Add(Cards.nineOf(suit));
            ExpectedCards.Add(Cards.tenOf(suit));
            ExpectedCards.Add(Cards.jackOf(suit));
            ExpectedCards.Add(Cards.queenOf(suit));
            ExpectedCards.Add(Cards.kingOf(suit));
        }

        Deck.populate();

        Assert.AreEqual(ExpectedCards, Deck.getCards());
    }
}